### PR TITLE
Adding first set of benchmarks and infrastructure for it

### PR DESCRIPTION
**Pull request by manodasanW**
- Setup a benchmark infrastructure using BenchmarkDotNet
- Benchmarks can be ran on NET Core 2.0, NET Core 3.1, and NET 5.0 using the CsWinRT projection via benchmark.cmd.
- To compare as a baseline, the same benchmarks can be built using the built-in WinMD support in NET Core 3.1 and ran via benchmark_winmd.cmd.
- Added benchmarks for QueryInterface scenarios which were used in the evaluation of the non default interface caching improvement.

### CODE REVIEW

File: Benchmarks/Benchmarks.csproj [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <PackageReference Include="Microsoft.Windows.SDK.Contracts" Version="10.0.18362.2005" Condition="$(UseWinmd) == true"/>
+    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.3" Condition="$(UseWinmd) == false"/>
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Projections\Windows\Windows.csproj" Condition="$(UseWinmd) == false" />
+    <ProjectReference Include="..\Projections\Benchmark\Benchmark.csproj" Condition="$(UseWinmd) == false" />
+    <ProjectReference Include="..\TestWinRT\BenchmarkComponent\BenchmarkComponent.vcxproj" Condition="$(UseWinmd) == true" />
+
+    <!-- The Benchmark / BenchmarkCompnent reference doesn't seem to bring over the dll when building with dotnet cli, so including both dll and winmd as includes. -->
```
Comment (Scottj1s) [inline Benchmarks/Benchmarks.csproj:25]: right - dotnet cli builds don't support vcxproj references

File: Benchmarks/Program.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+            public CustomConfig()
+            {
+                // Test CsWinRT projection
+                var job = Job.Default
+                    .WithPlatform(BenchmarkDotNet.Environments.Platform.X64)
+                    .WithArguments(
+                        new Argument[] {
+                            new MsBuildArgument("/p:platform=x64"),
+                            new MsBuildArgument("/p:MsAppxPackageTargets=%temp%\\EmptyMsAppxPackage.Targets")
```
Comment (Scottj1s) [inline Benchmarks/Program.cs:30]: This isn't needed any more - I removed it from the pipeline and build.cmd (so you won't have a %temp%\EmptyMsAppxPackage.Targets anyway)

File: Projections/Benchmark/Benchmark.csproj [hunk 3]
```diff
@@ -0,0 +1,66 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net5.0</TargetFrameworks>
+    <Platforms>x64;x86</Platforms>
+    <TargetPlatformVersion>10.0.18362.0</TargetPlatformVersion>
```
Comment (Scottj1s) [inline Projections/Benchmark/Benchmark.csproj:6]: redundant with Directory.Build.Props, if you want a single place to update all

File: README.md [hunk 4]
```diff
@@ -117,12 +117,16 @@ The **/TestComponentCSharp** folder contains an implementation of a WinRT test c
...
 ## /Projections
 
-The **/Projections** folder contains several projects for generating and building projections from the Windows SDK, WinUI, and Test metadata (produced by the TestWinRT and TestComponentCSharp projects).
+The **/Projections** folder contains several projects for generating and building projections from the Windows SDK, WinUI, Benchmark (produced by the BenchmarkComponent project), and Test metadata (produced by the TestWinRT and TestComponentCSharp projects).
 
 ## /UnitTest
 
 The **/UnitTest** folder contains unit tests for validating the Windows SDK, WinUI, and Test projections generated above.  All pull requests should ensure that this project executes without errors.
 
+## /Benchmarks
```
Comment (Scottj1s) [inline README.md:126]: thanks!

File: Projections/Benchmark/Benchmark.csproj (hunk 3 above)
Comment (manodasanW) [reply Projections/Benchmark/Benchmark.csproj]: Removed it for all projection csprojs.

File: Benchmarks/Program.cs (hunk 2 above)
Comment (manodasanW) [reply Benchmarks/Program.cs]: Yea it doesn't seem to be needed.  Removed it.

**Pull request merged**